crypto/s2n_hmac.c

```diff
@@ -192,9 +192,9 @@ int s2n_hmac_reset(struct s2n_hmac_state *state)
     return 0;
 }
 
-int s2n_hmac_digest_verify(const void *a, uint32_t alen, const void *b, uint32_t blen)
+int s2n_hmac_digest_verify(const void *a, const void *b, uint32_t len)
 {
-    return 0 - (!s2n_constant_time_equals(a, b, alen) | !!(alen - blen));
+    return 0 - !s2n_constant_time_equals(a, b, len);
```

Comment:
I'm not sure I follow how this helps. How do we know that the lengths of a and b are the same at this point? 

Would it be simpler to make the code do:


    if (alen != blen) {
        return -1;
    }
    
    return 0 -  !s2n_constant_time_equals(a, b, alen);

?